packages/core/src/components/section/section.tsx

```diff
@@ -158,38 +161,36 @@ export const Section: React.FC<SectionProps> = React.forwardRef((props, ref) =>
             })}
             elevation={elevation}
             ref={ref}
+            role="region"
+            aria-labelledby={sectionTitleId}
             {...htmlProps}
+            id={sectionId}
         >
             {title && (
                 <div
                     role={collapsible ? "button" : undefined}
                     aria-pressed={collapsible ? isCollapsed : undefined}
+                    aria-expanded={collapsible ? isCollapsed : undefined}
+                    aria-controls={sectionId}
                     className={classNames(Classes.SECTION_HEADER, {
                         [Classes.INTERACTIVE]: collapsible,
                     })}
-                    onClick={collapsible != null ? toggleIsCollapsed : undefined}
+                    onClick={collapsible ? toggleIsCollapsed : undefined}
                 >
-                    {isHeaderLeftContainerVisible && (
```

Comment:
`isHeaderLeftContainerVisible` is not needed, does nothing, because of line 169 `title &&` , `isHeaderLeftContainerVisible` is always true